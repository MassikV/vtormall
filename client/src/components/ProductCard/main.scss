.home-cards{
  min-width: 300px;
  width: 100%;
  @media (min-width: 768px) {
   max-width: 362px;
  }
  font-family: Roboto, sans-serif;
  max-height: 480px;
  position: relative;
}
.productCard-star{
  position: absolute;
  top: 5px;
  right: 10px;
}