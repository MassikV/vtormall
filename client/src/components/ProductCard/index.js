//@flow

import * as React from 'react';
import { Card } from 'antd';
import './main.scss';
import FavoriteService from '../FavoriteService';

const handleTypeTranslate = (type: string) => {
  switch (type) {
    case 'buy':
      return 'купівля';
    case 'sell':
      return 'продаж';
    case 'free':
      return 'безкоштовно';
    default:
      return type;
  }
};

const handleCategoryTranslate = (category: string) => {
  switch (category) {
    case 'glass':
      return 'скло';
    case 'plastic':
      return 'пластик';
    case 'paper':
      return 'папір';
    case 'metal':
      return 'метал';
    default:
      return category;
  }
};

type ProductCardTypes = {
  price: number,
  image: string,
  category: string,
  title: string,
  currency: string,
  type: string,
  favorite: boolean,
};

function ProductCard({
  id = null,
  price,
  image,
  category,
  title,
  currency,
  type,
  favorite = false,
  userToken = false,
}: ProductCardTypes): React.Node {
  return (
    <Card
      hoverable
      className="home-cards"
      bodyStyle={{ paddingLeft: '10px', paddingRight: '10px' }}
      cover={<img style={{ height: '300px', objectFit: 'cover' }} alt="product" src={image} />}>
      {id !== null && (
        <div className="productCard-star">
          <FavoriteService fill={favorite} token={userToken} id={id} />
        </div>
      )}
      <h4 className="fs-6 mb-3">{title}</h4>
      <p>
        тип оголошення: <span className="fw-bolder">{handleTypeTranslate(type)}</span>
      </p>
      <p className="fst-italic">
        ціна: {price.toLocaleString('de-DE', { style: 'currency', currency: currency })}
      </p>
      <Card.Meta description={handleCategoryTranslate(category)} />
    </Card>
  );
}

export default ProductCard;
