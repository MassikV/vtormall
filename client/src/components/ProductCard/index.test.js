import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ProductCard from './index';

describe('ProductCard component', () => {
  it('render ProductCard component', () => {
    render(
      <ProductCard
        title="Test"
        price="Test"
        image="https://upload.wikimedia.org/wikipedia/commons/b/bd/Test.svg"
        category="Test"
        currency="EUR"
        type="sell"
      />
    );
  });
  it('should correctly render content', () => {
    render(
      <ProductCard
        title="Test title in Product Card"
        price={22345}
        image="https://upload.wikimedia.org/wikipedia/commons/b/bd/Test.svg"
        category="Test category in Product Card"
        currency="EUR"
        type="sell"
      />
    );
    const titleElement = screen.getByText('Test title in Product Card');
    expect(titleElement).toBeInTheDocument();
    const priceElement = screen.getByText('ціна: 22.345,00 €');
    expect(priceElement).toBeInTheDocument();
    const imageElement = screen.getByAltText('product');
    expect(imageElement.src).toContain(
      'https://upload.wikimedia.org/wikipedia/commons/b/bd/Test.svg'
    );
    const categoryElement = screen.getByText('Test category in Product Card');
    expect(categoryElement).toBeInTheDocument();
    const typeElement = screen.getByText('продаж');
    expect(typeElement).toBeInTheDocument();
  });
  it('should render correctly ProductCard snapshot', () => {
    expect(
      render(
        <ProductCard
          title="Test title in Product Card"
          price={22345}
          image="https://upload.wikimedia.org/wikipedia/commons/b/bd/Test.svg"
          category="Test category in Product Card"
          currency="EUR"
          type="sell"
        />
      )
    ).toMatchSnapshot();
  });
});
