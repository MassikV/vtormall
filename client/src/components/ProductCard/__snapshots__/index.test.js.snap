// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard component should render correctly ProductCard snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ant-card ant-card-bordered ant-card-hoverable home-cards css-dev-only-do-not-override-26rdvq"
      >
        <div
          class="ant-card-cover"
        >
          <img
            alt="product"
            crossorigin="anonymous"
            src="https://upload.wikimedia.org/wikipedia/commons/b/bd/Test.svg"
            style="height: 300px; object-fit: cover;"
          />
        </div>
        <div
          class="ant-card-body"
          style="padding-left: 10px; padding-right: 10px;"
        >
          <h4
            class="fs-6 mb-3"
          >
            Test title in Product Card
          </h4>
          <p>
            тип оголошення: 
            <span
              class="fw-bolder"
            >
              продаж
            </span>
          </p>
          <p
            class="fst-italic"
          >
            ціна: 
            22.345,00 €
          </p>
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-description"
              >
                Test category in Product Card
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ant-card ant-card-bordered ant-card-hoverable home-cards css-dev-only-do-not-override-26rdvq"
    >
      <div
        class="ant-card-cover"
      >
        <img
          alt="product"
          crossorigin="anonymous"
          src="https://upload.wikimedia.org/wikipedia/commons/b/bd/Test.svg"
          style="height: 300px; object-fit: cover;"
        />
      </div>
      <div
        class="ant-card-body"
        style="padding-left: 10px; padding-right: 10px;"
      >
        <h4
          class="fs-6 mb-3"
        >
          Test title in Product Card
        </h4>
        <p>
          тип оголошення: 
          <span
            class="fw-bolder"
          >
            продаж
          </span>
        </p>
        <p
          class="fst-italic"
        >
          ціна: 
          22.345,00 €
        </p>
        <div
          class="ant-card-meta"
        >
          <div
            class="ant-card-meta-detail"
          >
            <div
              class="ant-card-meta-description"
            >
              Test category in Product Card
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
